pkg/machines/components/vm/bootOrderModal.jsx

```diff
@@ -0,0 +1,301 @@
+        } else if (device.device.type === "scsi") {
+            addOptional(additionalInfo, device.device.type, _("Type"));
+            addOptional(additionalInfo, device.device.source.address.bus, _("Bus"));
+            addOptional(additionalInfo, device.device.source.address.target, _("Target"));
+            addOptional(additionalInfo, device.device.source.address.unit, _("Unit"));
+        } else if (device.device.type === "scsi_host") {
+            addOptional(additionalInfo, device.device.type, _("Type"));
+            addOptional(additionalInfo, device.device.source.protocol, _("Protocol"));
+            addOptional(additionalInfo, device.device.source.wwpn, _("WWPN"));
+        } else if (device.device.type === "mdev") {
+            addOptional(additionalInfo, device.device.type, _("Type"));
+            addOptional(additionalInfo, nodeDev.xml.device.capability.type._attributes.id, _("Type ID"));
+        }
+    }
+
+    let upArrow;
+    let downArrow;
+    if (upDisabled)
+        upArrow = (<Button disabled><Icon id={`${idPrefix}-up`} type="fa" name="angle-up" /></Button>);
+    else
+        upArrow = (<Button onClick={moveUp}><Icon id={`${idPrefix}-up`} type="fa" name="angle-up" /></Button>);
+
+    if (downDisabled)
+        downArrow = (<Button disabled><Icon id={`${idPrefix}-down`} type="fa" name="angle-down" /></Button>);
+    else
+        downArrow = (<Button onClick={moveDown}><Icon id={`${idPrefix}-down`} type="fa" name="angle-down" /></Button>);
+
+    const actions = (
+        <ButtonGroup>
+            {upArrow}
+            {downArrow}
+        </ButtonGroup>
+    );
+
+    return (
+        <ListViewItem
+            id={`${idPrefix}-device-row-${index}`}
+            checkboxInput={<input type="checkbox" checked={device.checked} onChange={onToggle} />}
+            heading={heading}
+            additionalInfo={additionalInfo}
+            actions={actions}
+        />
+    );
+};
+
+export class BootOrderModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            devices: getBootOrderDevices(props.vm),
+            dialogError: undefined,
+            dialogErrorDetail: undefined,
+        };
+        this.dialogErrorSet = this.dialogErrorSet.bind(this);
+        this.close = props.close;
+        this.save = this.save.bind(this);
+        this.onToggleDevice = this.onToggleDevice.bind(this);
+        this.move = this.move.bind(this);
+    }
+
+    dialogErrorSet(text, detail) {
+        this.setState({ dialogError: text, dialogErrorDetail: detail });
+    }
+
+    save() {
+        const { dispatch, vm } = this.props;
+        const devices = this.state.devices.filter((d) => d.checked);
+
+        dispatch(changeBootOrder({
+            vm,
+            devices,
+        }))
+                .fail((exc) => {
+                    this.dialogErrorSet(_("Boot order settings could not be saved"), exc.message);
```

Comment:
Also then there is a single liner function handler you don't need the curly brackets. Just remove the brackets and bring it to the previous line.